Description: make kw installation closer to unix standard
Author: Rodrig Carvalho <rodrigorsdc@gmail.com>

This patch create a Makefile and modify files in order to
make installation of kw closer to unix standard

---
The information above should follow the Patch Tagging Guidelines, please
checkout http://dep.debian.net/deps/dep3/ to learn about the format. Here
are templates for supplementary fields that you might want to add:

Origin: <vendor|upstream|other>, <url of original patch>
Bug: <url in upstream bugtracker>
Bug-Debian: https://bugs.debian.org/<bugnumber>
Bug-Ubuntu: https://launchpad.net/bugs/<bugnumber>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: <name and email of someone who approved the patch>
Last-Update: 2019-12-12

Index: kw-20191112/Makefile
===================================================================
--- /dev/null
+++ kw-20191112/Makefile
@@ -0,0 +1,42 @@
+PREFIX = ${DESTDIR}/usr
+STDDIR = ${PREFIX}/share/kw
+MANDIR = ${PREFIX}/share/man
+ACDIR = ${PREFIX}/share/bash-completion/completions
+
+all:
+	@echo "kworkflow is a shell script program. Try \"make install\""
+
+bash-autocomplete:
+	@install -v -d ${ACDIR}
+	@install -v -m 0755 -T src/bash_autocomplete.sh ${ACDIR}/kw
+
+install: bash-autocomplete
+	@install -v -d ${MANDIR}/man1
+	@install -v -d /etc/skel/.config/kw
+	rst2man < documentation/man/kw.rst > ${MANDIR}/man1/kw.1
+	@install -v -d ${STDDIR}/src
+	@install -v -d ${STDDIR}/sounds
+	@install -v -d ${STDDIR}/etc
+	@install -v -d ${STDDIR}/deploy_rules
+	@install -v -m 0644 src/* ${STDDIR}/src
+	@chmod 755 ${STDDIR}/src/kw.fish
+	@rm -v ${STDDIR}/src/bash_autocomplete.sh
+	@install -v -m 0644 sounds/* ${STDDIR}/sounds
+	@install -v -m 0644 etc/* ${STDDIR}/etc
+	@cp -vr deploy_rules ${STDDIR}/deploy_rules
+	@install -v kw ${STDDIR}
+
+remove-bash-autocomplete:
+	@rm -vrf \
+		${ACDIR}/kw
+
+uninstall: remove-bash-autocomplete
+	@rm -vrf \
+		${STDDIR} \
+		${MANDIR}/man1/kw.1
+
+clean:
+	@rm -vrf \
+		build/
+
+.PHONY: build-man install uninstall
Index: kw-20191112/etc/kworkflow.config
===================================================================
--- kw-20191112.orig/etc/kworkflow.config
+++ kw-20191112/etc/kworkflow.config
@@ -37,7 +37,7 @@ qemu_path_image=/home/USERKW/p/virty.qco
 alert=n
 
 # Command to run for sound completion alert
-sound_alert_command=paplay INSTALLPATH/sounds/complete.wav &
+sound_alert_command=paplay /usr/share/kw/sounds/complete.wav &
 
 # Command to run for visual completion alert
 # Note: You may use $COMMAND, which will be replaced by the kw command
Index: kw-20191112/kw
===================================================================
--- kw-20191112.orig/kw
+++ kw-20191112/kw
@@ -1,10 +1,13 @@
 #!/bin/bash
 
 # Set required variables
+CURDIR=$(dirname $(realpath "$0"))
+
+export CURDIR
+
 EASY_KERNEL_WORKFLOW=${EASY_KERNEL_WORKFLOW:-"kw"}
-src_script_path=${src_script_path:-"$HOME/.config/$EASY_KERNEL_WORKFLOW/src"}
-external_script_path=${external_script_path:-"$HOME/.config/$EASY_KERNEL_WORKFLOW/external"}
-etc_files_path=${etc_files_path:-"$HOME/.config/$EASY_KERNEL_WORKFLOW/etc"}
+src_script_path=${src_script_path:-"$CURDIR/src"}
+etc_files_path=${etc_files_path:-"$CURDIR/etc"}
 config_files_path=${config_files_path:-"$HOME/.config/$EASY_KERNEL_WORKFLOW"}
 
 # Export external variables required by kworkflow
@@ -168,4 +171,11 @@ function kw()
   esac
 }
 
+# If user home doesn't contain config files
+if [ ! -d "$HOME/.config/kw" ]; then
+    install -d "$HOME/.config/kw"
+    cp -a "${CURDIR}/etc/." "${config_files_path}"
+    sed -i "s/USERKW/$USER/g" "${config_files_path}/kworkflow.config"
+fi
+
 kw $@
Index: kw-20191112/src/commons.sh
===================================================================
--- kw-20191112.orig/src/commons.sh
+++ kw-20191112/src/commons.sh
@@ -59,7 +59,7 @@ function load_configuration()
   local local_config_path=$PWD/kworkflow.config
 
   # First, load the global configuration
-  parse_configuration $etc_files_path/kworkflow.config
+  parse_configuration $config_files_path/kworkflow.config
 
   # Second, check if has a local file and override values
   if [ -f $local_config_path ] ; then
Index: kw-20191112/src/vm.sh
===================================================================
--- kw-20191112.orig/src/vm.sh
+++ kw-20191112/src/vm.sh
@@ -101,7 +101,7 @@ function vm_ssh
 
 function vm_prepare
 {
-  local path_ansible=$HOME/.config/kw/deploy_rules/
+  local path_ansible=$CURDIR/deploy_rules/
   local current_path=$PWD
   local ret=0
   say "Deploying with Ansible, this will take some time"
